From 473a82572ebe65582627947e9ea750cc0079418a Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Sat, 26 Apr 2014 18:44:39 +0200
Subject: [PATCH 10/28] A very basic unittest for TagListJob

... which doesn't actually test anything relating to baloo or tags ;-)
---
 src/file/lib/autotests/CMakeLists.txt     | 16 +++++++++++++
 src/file/lib/autotests/taglistjobtest.cpp | 36 ++++++++++++++++++++++++++++++
 src/file/lib/autotests/taglistjobtest.h   | 37 +++++++++++++++++++++++++++++++
 3 files changed, 89 insertions(+)
 create mode 100644 src/file/lib/autotests/taglistjobtest.cpp
 create mode 100644 src/file/lib/autotests/taglistjobtest.h

diff --git a/src/file/lib/autotests/CMakeLists.txt b/src/file/lib/autotests/CMakeLists.txt
index 31dc368..95a0916 100644
--- a/src/file/lib/autotests/CMakeLists.txt
+++ b/src/file/lib/autotests/CMakeLists.txt
@@ -49,3 +49,19 @@ target_link_libraries(modifyjobtest
 # temporary files they need (as XattrDetector blacklists /tmp by default).
 set_target_properties(fetchjobtest PROPERTIES COMPILE_FLAGS -DBUILDDIR="\\"${CMAKE_CURRENT_BINARY_DIR}/\\"")
 set_target_properties(modifyjobtest PROPERTIES COMPILE_FLAGS -DBUILDDIR="\\"${CMAKE_CURRENT_BINARY_DIR}/\\"")
+
+#
+# TagList Job
+#
+set(taglistjobtest_SRC taglistjobtest.cpp)
+kde4_add_unit_test(taglistjobtest NOGUI ${taglistjobtest_SRC})
+target_link_libraries(taglistjobtest
+  ${QT_QTTEST_LIBRARY}
+  ${KDE4_KDECORE_LIBS}
+  #${KDE4_SOLID_LIBS}
+  ${QT_QTSQL_LIBRARY}
+  ${XAPIAN_LIBRARIES}
+  ${QJSON_LIBRARIES}
+  baloocore
+  baloofiles
+)
diff --git a/src/file/lib/autotests/taglistjobtest.cpp b/src/file/lib/autotests/taglistjobtest.cpp
new file mode 100644
index 0000000..1d7f34b
--- /dev/null
+++ b/src/file/lib/autotests/taglistjobtest.cpp
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2014  David Faure <faure@kde.org>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#include "taglistjobtest.h"
+#include "taglistjob.h"
+
+#include "qtest_kde.h"
+
+using namespace Baloo;
+
+void TagListJobTest::testExec()
+{
+    // A very basic test to ensure that autodeletion works, while debugging a gwenview issue.
+    QPointer<Baloo::TagListJob> job = new Baloo::TagListJob();
+    job->exec();
+    qApp->sendPostedEvents(job, QEvent::DeferredDelete);
+    QVERIFY(job == 0);
+}
+
+QTEST_KDEMAIN_CORE(TagListJobTest)
diff --git a/src/file/lib/autotests/taglistjobtest.h b/src/file/lib/autotests/taglistjobtest.h
new file mode 100644
index 0000000..09123c2
--- /dev/null
+++ b/src/file/lib/autotests/taglistjobtest.h
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2014  David Faure <faure@kde.org>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#ifndef FILEMODIFYJOBTEST_H
+#define FILEMODIFYJOBTEST_H
+
+#include <QObject>
+
+namespace Baloo {
+
+class TagListJobTest : public QObject
+{
+    Q_OBJECT
+
+private Q_SLOTS:
+    void testExec();
+};
+
+}
+
+#endif
-- 
1.9.0

